File: src/main/java/com/ject/studytrip/trip/application/dto/TripReportDetail.java
package com.ject.studytrip.trip.application.dto;

import com.ject.studytrip.studylog.application.dto.StudyLogSliceInfo;

public record TripReportDetail(TripReportInfo tripReportInfo, StudyLogSliceInfo studyLogSliceInfo) {
    public static TripReportDetail from(
            TripReportInfo tripReportInfo, StudyLogSliceInfo studyLogSliceInfo) {
        return new TripReportDetail(tripReportInfo, studyLogSliceInfo);
    }
}

File: src/main/kotlin/com/ject/studytrip/trip/application/dto/TripReportDetail.kt
package com.ject.studytrip.trip.application.dto

import com.ject.studytrip.studylog.application.dto.StudyLogSliceInfo

class TripReportDetail(
    val tripReportInfo: TripReportInfo,
    val studyLogSliceInfo: StudyLogSliceInfo,
) {
    companion object {
        @JvmStatic
        fun from(
            tripReportInfo: TripReportInfo,
            studyLogSliceInfo: StudyLogSliceInfo,
        ) = TripReportDetail(tripReportInfo, studyLogSliceInfo)
    }
}
